src/main/scala/com/rbmhtechnology/eventuate/ReplicationProtocol.scala

```diff
@@ -134,6 +152,17 @@ object ReplicationProtocol {
   case class ReplicationRead(fromSequenceNr: Long, maxNumEvents: Int, filter: ReplicationFilter, targetLogId: String, replicator: ActorRef, currentTargetVersionVector: VectorTime) extends Format
 
   /**
+   * Marks all possible responses to a [[ReplicationRead]] request
+   */
+  trait ReplicationReadResponse extends Format
+
+  /**
+   * Wraps a [[ReplicationReadResponse]] and includes the id of the source log.
+   * This id is used to track replication progress per log in case of endpoint recovery.
+   */
+  case class ReplicationReadResponseEnvelope(payload: ReplicationReadResponse, sourceLogId: String)
```

Comment:
Should you decide to revert the [commented parts](https://github.com/RBMHTechnology/eventuate/pull/162#commitcomment-14801908) of #162 (which would introduce `ReplicationReadResponseEnvelope` again), please extend `Format` so that the `ReplicationProtocolSerializer` is used (instead of default Java serialization).